assignment-solutions/commands.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"strconv"
+)
+
+var file string = "history.txt"
+var i int = 0
+
+func displayPrompt() {
+	// get the current user
+	currentUser, _ := user.Current()
+
+	// get the hostname
+	hostname, _ := os.Hostname()
+
+	// get the current working directory
+	pwd := showPresentWorkingDir()
+
+	prompt := currentUser.Username + "@" + hostname + " ~" + pwd + "$ "
+	fmt.Print("\n" + prompt)
+}
+
+func executeCommand(cmd string, args []string) {
+	fmt.Print("\nExecuting command :", cmd+"\n")
+	fmt.Print("-----------------------\n")
+
+	// add the command into file
+	i++
+	updateHistoryFile(strconv.Itoa(i) + " " + cmd)
+
+	if len(cmd) == 0 {
+		fmt.Println("Error : not received any command, exit!")
+	}
+
+	switch cmd {
+	case "ls":
+		showFilesAndDirectories(args)
+
+	case "cd":
+		processChangeWorkingDir(args)
+
+	case "pwd":
+		fmt.Println(cmd, ": Current working directory : ", showPresentWorkingDir())
+
+	case "history":
+		// read file here
+		data, err := ioutil.ReadFile(file)
+		if err != nil {
+			fmt.Println("Error : Unable to read a file!")
+		}
+		fmt.Println(string(data))
+
+	case "exit":
+		fmt.Println("\nExiting from the shell. Thank you!")
+		os.Remove(file)
+		os.Exit(0)
+
+	default:
+		fmt.Println(cmd, " command not found..!")
+	}
+}
+
+// Display current working directory
+func showPresentWorkingDir() string {
+
+	dir, err := os.Getwd()
+
+	if err != nil {
+		fmt.Println("Error : ", err)
+	}
+
+	return dir
+}
+
+// Change the working directory
+func processChangeWorkingDir(args []string) {
+	if len(args) == 0 {
+		fmt.Println("Error : Invalid command argument!")
+	}
```

Comment:
If no args you should go to home directory.

Reply:
If more than one args then return error

Reply:
Sure